packages/@react-aria/overlays/src/useModalOverlay.ts

```diff
@@ -67,3 +68,120 @@ export function useModalOverlay(props: AriaModalOverlayProps, state: OverlayTrig
     underlayProps
   };
 }
+
+function hideElementsBehind(element: Element, root = document.body) {
+  // TODO: automatically determine root based on parent stacking context of element?
+  let roots = getStackingContextRoots(root);
```

Comment:
this could be potentially really bad, calling getComputedStyle on every element on the entire page, though unlikely

I guess it only happens on mount of a dialog or something along those lines

We can't really keep a single, on page load data structure with all of them and update as the dom updates because css could change, and it'd involve mutation observers on way too many things which would cause a performance hit for other reasons

it'd be nice if we could just compare two elements traversing upwards instead, but that'd be really hard for the initial hide